editor/editor_dock_manager.cpp

```diff
@@ -149,17 +149,56 @@ void EditorDockManager::_update_layout() {
 	}
 	EditorNode::get_singleton()->edit_current();
 	dock_context_popup->docks_updated();
+	_update_docks_menu();
 	EditorNode::get_singleton()->save_editor_layout_delayed();
 }
 
+void EditorDockManager::_update_docks_menu() {
+	docks_menu->clear();
+	docks_menu->reset_size();
+
+	const Ref<Texture2D> icon = docks_menu->get_editor_theme_icon(SNAME("Window"));
+	const Color open_icon_color_mod = Color(1, 1, 1);
+	const Color closed_icon_color_mod = Color(0.5, 0.5, 0.5);
```

Comment:
This will look strange on a light theme (here, the first dock is closed and other docks are open):

![image](https://github.com/godotengine/godot/assets/180032/4bd68a3d-9e2c-46b8-a939-58fc2f3fec8c)

To avoid this, I suggest multiplying opacity by 0.5 instead of multiplying the color.